analysers/experiments/constructDuration.py

```diff
@@ -0,0 +1,78 @@
+import sys
+import json
+import io
+import gzip
+import uuid
+import math
+
+from datetime import timedelta
+
+from pyspark_cassandra import CassandraSparkContext
+from pyspark_cassandra import RowFormat
+from pyspark import SparkConf
+
+def createQuery(CassandraRDD, experimentID):
+    from commons import computeExperimentMetrics, computeModeMinMax
+    
+    queries = []
+    
+    constructs = CassandraRDD.map(lambda a: a["construct_name"]).distinct().collect()
```

Comment:
Select the distinct couple of `construct_name` AND `construct_type`, because you already have all the information on the actual existing couple, then you can avoid the next O(N^2) loop and just iterate linearly over the couple.